src/esp/gfx/Viewer.h

```diff
@@ -37,17 +39,42 @@ class Viewer : public Magnum::Platform::Application {
   void mouseScrollEvent(MouseScrollEvent& event) override;
   void keyPressEvent(KeyEvent& event) override;
 
+  void addObject();
+
   assets::ResourceManager resourceManager_;
+  assets::PhysicsManager physicsManager_;
   scene::SceneManager sceneManager_;
   std::vector<int> sceneID_;
   scene::SceneNode* agentBodyNode_ = nullptr;
   scene::SceneNode* cameraNode_ = nullptr;
+  
+  scene::SceneNode* navSceneNode_ = nullptr;
+  scene::SceneNode* objNode_ = nullptr;
+
+  std::vector<scene::SceneNode*> allObjNodes_;
+  //physics::BulletRigidObject* physicalObj_;
+  //physics::BulletRigidObject* physicalObj_;
+
+  scene::SceneGraph* sceneGraph;
+  scene::SceneNode* rootNode;
+
   RenderCamera* renderCamera_ = nullptr;
   nav::PathFinder::ptr pathfinder_;
   scene::ObjectControls controls_;
   Magnum::Vector3 previousPosition_;
 
   bool computeActionPath_;
+  bool enablePhysics_;
+  bool surreal_mesh = false;
+  bool replica_mesh = false;
+  bool vangoth_mesh = true;
+  bool castle_mesh = false;
+
```

Comment:
These are very hacky way to specify initial camera location, based on which mesh we're using.